Koronavilkku/Koronavilkku/Repository/ExposureRepository.swift

```diff
@@ -14,19 +24,96 @@ protocol ExposureRepository {
+    private var detectionTask: AnyCancellable?
+    
+    @Published
+    private var detectionRunning = false
     
+    lazy var detectionStatus: AnyPublisher<DetectionStatus, Never> = {
+        let isDelayed = timeFromLastCheck.map { interval -> Bool in
+            guard let interval = interval else { return false }
+            return interval <= 0 - Self.manualCheckThreshold
+        }
+        
+        let isDisabled = LocalStore.shared.$uiStatus.$wrappedValue.map { uiStatus -> Bool in
+            switch uiStatus {
+            case .apiDisabled, .off:
+                return true
+            default:
+                return false
+            }
+        }
+        
+        return $detectionRunning.combineLatest(isDelayed, isDisabled) { running, delayed, disabled in
+            switch true {
+            case disabled:
+                return .disabled
+            case running:
+                return .detecting
+            default:
+                return .idle(delayed: delayed)
+            }
+        }.eraseToAnyPublisher()
+    }()
+        
+    lazy var timeFromLastCheck: AnyPublisher<TimeInterval?, Never> = {
+        // Create a timer based on the current exposure detection date
+        LocalStore.shared.$dateLastPerformedExposureDetection.$wrappedValue.map { lastCheck -> AnyPublisher<TimeInterval?, Never> in
+            guard let lastCheck = lastCheck else {
+                return Just(nil).eraseToAnyPublisher()
+            }
+            
+            let publisher = Just(Date())
+            let timer = Timer.publish(every: 60, tolerance: 1, on: .main, in: .default).autoconnect()
+            
+            return publisher
+                .merge(with: timer)
+                .map { $0.distance(to: lastCheck) }
+                .eraseToAnyPublisher()
+        }
+        // Cancels the previous timer when the value changes to avoid mixed signals
+        .switchToLatest()
+        // Make sure the same value is broadcasted to every subscriber *and* that the current value is always published
+        .multicast(subject: CurrentValueSubject<TimeInterval?, Never>(nil))
+        .autoconnect()
+        .eraseToAnyPublisher()
+    }()
+
     init(exposureManager: ExposureManager, backend: Backend, storage: FileStorage) {
         self.exposureManager = exposureManager
         self.backend = backend
         self.storage = storage
     }
     
+    func runManualCheck() -> AnyPublisher<Bool, Never> {
+        var backgroundId: UIBackgroundTaskIdentifier! = nil
+        
+        backgroundId = UIApplication.shared.beginBackgroundTask() { [weak self] in
+            self?.detectionTask?.cancel()
+            UIApplication.shared.endBackgroundTask(backgroundId)
+            backgroundId = .invalid
+        }
+        
+        return Future { promise in
+            self.detectionRunning = true
+            self.detectionTask = BackgroundTaskForNotifications.execute { [weak self] success in
```

Comment:
Is it possible to start a manual check while a scheduled background task is already running, or vice versa?